#Study Understanding

##Study Objective

Proyek ini bertujuan untuk memahami bagaimana penggunaan media sosial mempengaruhi kesehatan mental, termasuk tingkat stres, kecemasan, dan depresi. Analisis ini akan mengeksplorasi hubungan antara durasi penggunaan media sosial, jenis platform yang digunakan, serta dampaknya terhadap kondisi psikologis individu.

## Assess Situation

Media sosial telah menjadi bagian besar dari kehidupan modern, tetapi dampaknya terhadap kesehatan mental masih menjadi perdebatan. Beberapa penelitian menunjukkan bahwa penggunaan media sosial yang berlebihan dapat meningkatkan risiko gangguan mental, sementara yang lain menemukan bahwa platform ini juga dapat memberikan dukungan sosial.  

Dataset yang digunakan adalah **Social Media and Mental Health**, yang berisi data mengenai kebiasaan penggunaan media sosial serta kondisi kesehatan mental responden. Analisis akan fokus pada bagaimana frekuensi penggunaan, jenis interaksi, dan faktor lainnya berhubungan dengan kesehatan mental.  

## Data Mining Goals  

- Mengidentifikasi pola hubungan antara intensitas penggunaan media sosial dan tingkat kesehatan mental.  
- Menemukan segmentasi pengguna berdasarkan durasi dan cara mereka menggunakan media sosial.  
- Mengeksplorasi faktor-faktor yang memperburuk atau memperbaiki dampak media sosial terhadap kesehatan mental.  
- Menentukan apakah ada hubungan antara jenis aktivitas di media sosial (scrolling pasif, interaksi aktif, dll.) dengan kesehatan mental.

## Project Plan

### Data Understanding  
- Mengeksplorasi dataset **Social Media and Mental Health** untuk memahami struktur data dan variabel yang tersedia.  
- Menganalisis distribusi data, tipe data, serta mengidentifikasi potensi permasalahan seperti missing values atau outlier.  
- Menentukan variabel utama yang akan dianalisis, seperti durasi penggunaan media sosial, tingkat interaksi, dan dampaknya terhadap kesehatan mental.  

### Data Preparation  
- Membersihkan data, menangani nilai yang hilang, serta melakukan transformasi atau encoding yang diperlukan.  
- Mengelompokkan data untuk keperluan analisis lebih lanjut, seperti segmentasi pengguna berdasarkan pola penggunaan media sosial.  

### Visualisasi  
- Membuat visualisasi untuk memahami pola hubungan antara variabel yang dianalisis.  
- Menggunakan grafik, heatmap, dan diagram hubungan untuk menampilkan korelasi antara durasi penggunaan media sosial dengan tingkat stres, kecemasan, dan depresi.  

### Dashboard  
- Membangun dashboard interaktif untuk menyajikan hasil utama dari analisis.  
- Dashboard akan menampilkan insight utama terkait dampak media sosial terhadap kesehatan mental dalam bentuk grafik dan metrik yang mudah dipahami.  

### Insight & Action  
- Menarik kesimpulan dari hasil analisis mengenai bagaimana media sosial mempengaruhi kesehatan mental.  
- Memberikan rekomendasi berbasis data, seperti pola penggunaan media sosial yang lebih sehat atau strategi untuk mengurangi dampak negatif dari media sosial terhadap kesejahteraan mental.

# DATA UNDERSTANDING

"Scroll tanpa Kontrol: bagaimana media sosial mempengaruhi kesehatan mental?"

Dataset yang saya gunakan dari kaggle.com

https://www.kaggle.com/datasets/souvikahmed071/social-media-and-mental-health

In [ ]:
import pandas as pd

file = "/content/smmh.csv"

df = pd.read_csv(file)
df

Timestamp  1. What is your age? 2. Gender  \
0     4/18/2022 19:18:47                  21.0      Male   
1     4/18/2022 19:19:28                  21.0    Female   
2     4/18/2022 19:25:59                  21.0    Female   
3     4/18/2022 19:29:43                  21.0    Female   
4     4/18/2022 19:33:31                  21.0    Female   
..                   ...                   ...       ...   
476   5/21/2022 23:38:28                  24.0      Male   
477    5/22/2022 0:01:05                  26.0    Female   
478   5/22/2022 10:29:21                  29.0    Female   
479   7/14/2022 19:33:47                  21.0      Male   
480  11/12/2022 13:16:50                  53.0      Male   

    3. Relationship Status 4. Occupation Status  \
0        In a relationship   University Student   
1                   Single   University Student   
2                   Single   University Student   
3                   Single   University Student   
4                   Single   University Student   
..                     ...                  ...   
476                 Single      Salaried Worker   
477                Married      Salaried Worker   
478                Married      Salaried Worker   
479                 Single   University Student   
480                Married      Salaried Worker   

    5. What type of organizations are you affiliated with?  \
0                                           University       
1                                           University       
2                                           University       
3                                           University       
4                                           University       
..                                                 ...       
476                                University, Private       
477                                         University       
478                                         University       
479                                         University       
480                                            Private       

    6. Do you use social media?  \
0                           Yes   
1                           Yes   
2                           Yes   
3                           Yes   
4                           Yes   
..                          ...   
476                         Yes   
477                         Yes   
478                         Yes   
479                         Yes   
480                         Yes   

    7. What social media platforms do you commonly use?  \
0    Facebook, Twitter, Instagram, YouTube, Discord...    
1    Facebook, Twitter, Instagram, YouTube, Discord...    
2              Facebook, Instagram, YouTube, Pinterest    
3                                  Facebook, Instagram    
4                         Facebook, Instagram, YouTube    
..                                                 ...    
476                       Facebook, Instagram, YouTube    
477                                  Facebook, YouTube    
478                                  Facebook, YouTube    
479  Facebook, Twitter, Instagram, YouTube, Discord...    
480                                  Facebook, YouTube    

    8. What is the average time you spend on social media every day?  \
0                                Between 2 and 3 hours                 
1                                    More than 5 hours                 
2                                Between 3 and 4 hours                 
3                                    More than 5 hours                 
4                                Between 2 and 3 hours                 
..                                                 ...                 
476                              Between 2 and 3 hours                 
477                              Between 1 and 2 hours                 
478                              Between 2 and 3 hours                 
479                              Between 2 and 3 hours                 
480           

In [ ]:
# Cek jumlah baris & kolom
df.shape

(481, 21)

Diatas adadalah hasil dari df.shape dengan baris 481 dan kolom 21

In [ ]:
df.dtypes

Timestamp                                                                                                                object
1. What is your age?                                                                                                    float64
2. Gender                                                                                                                object
3. Relationship Status                                                                                                   object
4. Occupation Status                                                                                                     object
5. What type of organizations are you affiliated with?                                                                   object
6. Do you use social media?                                                                                              object
7. What social media platforms do you commonly use?                                                                      object
8. What is the average time you spend on social media every day?                                                         object
9. How often do you find yourself using Social media without a specific purpose?                                          int64
10. How often do you get distracted by Social media when you are busy doing something?                                    int64
11. Do you feel restless if you haven't used Social media in a while?                                                     int64
12. On a scale of 1 to 5, how easily distracted are you?                                                                  int64
13. On a scale of 1 to 5, how much are you bothered by worries?                                                           int64
14. Do you find it difficult to concentrate on things?                                                                    int64
15. On a scale of 1-5, how often do you compare yourself to other successful people through the use of social media?      int64
16. Following the previous question, how do you feel about these comparisons, generally speaking?                         int64
17. How often do you look to seek validation from features of social media?                                               int64
18. How often do you feel depressed or down?                                                                              int64
19. On a scale of 1 to 5, how frequently does your interest in daily activities fluctuate?                                int64
20. On a scale of 1 to 5, how often do you face issues regarding sleep?                                                   int64
dtype: object

In [ ]:
df.describe()

1. What is your age?  \
count             481.00000   
mean               26.13659   
std                 9.91511   
min                13.00000   
25%                21.00000   
50%                22.00000   
75%                26.00000   
max                91.00000   

       9. How often do you find yourself using Social media without a specific purpose?  \
count                                         481.000000                                  
mean                                            3.553015                                  
std                                             1.096299                                  
min                                             1.000000                                  
25%                                             3.000000                                  
50%                                             4.000000                                  
75%                                             4.000000                                  
max                                             5.000000                                  

       10. How often do you get distracted by Social media when you are busy doing something?  \
count                                         481.000000                                        
mean                                            3.320166                                        
std                                             1.328137                                        
min                                             1.000000                                        
25%                                             2.000000                                        
50%                                             3.000000                                        
75%                                             4.000000                                        
max                                             5.000000                                        

       11. Do you feel restless if you haven't used Social media in a while?  \
count                                         481.000000                       
mean                                            2.588358                       
std                                             1.257059                       
min                                             1.000000                       
25%                                             2.000000                       
50%                                             2.000000                       
75%                                             3.000000                       
max                                             5.000000                       

       12. On a scale of 1 to 5, how easily distracted are you?  \
count                                         481.000000          
mean                                            3.349272          
std                                             1.175552          
min                                             1.000000          
25%                                             3.000000          
50%                                             3.000000          
75%                                             4.000000          
max                                             5.000000          

       13. On a scale of 1 to 5, how much are you bothered by worries?  \
count                                         481.000000                 
mean                                            3.559252                 
std                                             1.283356                 
min                                             1.000000                 
25%                                             3.000000                 
50%                                             4.000000                 
75%                                             5.000000                 
max                                             5.000000                 

       14. Do you find it difficult to co

Diatas adalah Statistik Deskriptif dari dataset yang saya gunakan.

# DATA PREPARATION

"Scroll tanpa Kontrol: bagaimana media sosial mempengaruhi kesehatan mental?"

Dataset yang saya gunakan dari kaggle.com

https://www.kaggle.com/datasets/souvikahmed071/social-media-and-mental-health

In [ ]:
import pandas as pd

file = "/content/smmh.csv"

df = pd.read_csv(file)
df

Timestamp  1. What is your age? 2. Gender  \
0     4/18/2022 19:18:47                  21.0      Male   
1     4/18/2022 19:19:28                  21.0    Female   
2     4/18/2022 19:25:59                  21.0    Female   
3     4/18/2022 19:29:43                  21.0    Female   
4     4/18/2022 19:33:31                  21.0    Female   
..                   ...                   ...       ...   
476   5/21/2022 23:38:28                  24.0      Male   
477    5/22/2022 0:01:05                  26.0    Female   
478   5/22/2022 10:29:21                  29.0    Female   
479   7/14/2022 19:33:47                  21.0      Male   
480  11/12/2022 13:16:50                  53.0      Male   

    3. Relationship Status 4. Occupation Status  \
0        In a relationship   University Student   
1                   Single   University Student   
2                   Single   University Student   
3                   Single   University Student   
4                   Single   University Student   
..                     ...                  ...   
476                 Single      Salaried Worker   
477                Married      Salaried Worker   
478                Married      Salaried Worker   
479                 Single   University Student   
480                Married      Salaried Worker   

    5. What type of organizations are you affiliated with?  \
0                                           University       
1                                           University       
2                                           University       
3                                           University       
4                                           University       
..                                                 ...       
476                                University, Private       
477                                         University       
478                                         University       
479                                         University       
480                                            Private       

    6. Do you use social media?  \
0                           Yes   
1                           Yes   
2                           Yes   
3                           Yes   
4                           Yes   
..                          ...   
476                         Yes   
477                         Yes   
478                         Yes   
479                         Yes   
480                         Yes   

    7. What social media platforms do you commonly use?  \
0    Facebook, Twitter, Instagram, YouTube, Discord...    
1    Facebook, Twitter, Instagram, YouTube, Discord...    
2              Facebook, Instagram, YouTube, Pinterest    
3                                  Facebook, Instagram    
4                         Facebook, Instagram, YouTube    
..                                                 ...    
476                       Facebook, Instagram, YouTube    
477                                  Facebook, YouTube    
478                                  Facebook, YouTube    
479  Facebook, Twitter, Instagram, YouTube, Discord...    
480                                  Facebook, YouTube    

    8. What is the average time you spend on social media every day?  \
0                                Between 2 and 3 hours                 
1                                    More than 5 hours                 
2                                Between 3 and 4 hours                 
3                                    More than 5 hours                 
4                                Between 2 and 3 hours                 
..                                                 ...                 
476                              Between 2 and 3 hours                 
477                              Between 1 and 2 hours                 
478                              Between 2 and 3 hours                 
479                              Between 2 and 3 hours                 
480           

1. Missing Values

In [ ]:
print((df.isna().sum() / len(df)) * 100)

Timestamp                                                                                                               0.000000
1. What is your age?                                                                                                    0.000000
2. Gender                                                                                                               0.000000
3. Relationship Status                                                                                                  0.000000
4. Occupation Status                                                                                                    0.000000
5. What type of organizations are you affiliated with?                                                                  6.237006
6. Do you use social media?                                                                                             0.000000
7. What social media platforms do you commonly use?                                              

2. Duplikat

In [ ]:
df[df.duplicated()]

Empty DataFrame
Columns: [Timestamp, 1. What is your age?, 2. Gender, 3. Relationship Status, 4. Occupation Status, 5. What type of organizations are you affiliated with?, 6. Do you use social media?, 7. What social media platforms do you commonly use?, 8. What is the average time you spend on social media every day?, 9. How often do you find yourself using Social media without a specific purpose?, 10. How often do you get distracted by Social media when you are busy doing something?, 11. Do you feel restless if you haven't used Social media in a while?, 12. On a scale of 1 to 5, how easily distracted are you?, 13. On a scale of 1 to 5, how much are you bothered by worries?, 14. Do you find it difficult to concentrate on things?, 15. On a scale of 1-5, how often do you compare yourself to other successful people through the use of social media?, 16. Following the previous question, how do you feel about these comparisons, generally speaking?, 17. How often do you look to seek validation from features of social media?, 18. How often do you feel depressed or down?, 19. On a scale of 1 to 5, how frequently does your interest in daily activities fluctuate?, 20. On a scale of 1 to 5, how often do you face issues regarding sleep?]
Index: []

[0 rows x 21 columns]

In [ ]:
df.columns

Index(['Timestamp', '1. What is your age?', '2. Gender',
       '3. Relationship Status', '4. Occupation Status',
       '5. What type of organizations are you affiliated with?',
       '6. Do you use social media?',
       '7. What social media platforms do you commonly use?',
       '8. What is the average time you spend on social media every day?',
       '9. How often do you find yourself using Social media without a specific purpose?',
       '10. How often do you get distracted by Social media when you are busy doing something?',
       '11. Do you feel restless if you haven't used Social media in a while?',
       '12. On a scale of 1 to 5, how easily distracted are you?',
       '13. On a scale of 1 to 5, how much are you bothered by worries?',
       '14. Do you find it difficult to concentrate on things?',
       '15. On a scale of 1-5, how often do you compare yourself to other successful people through the use of social media?',
       '16. Following the previous question, 

In [ ]:
df.dtypes

Timestamp                                                                                                                object
1. What is your age?                                                                                                    float64
2. Gender                                                                                                                object
3. Relationship Status                                                                                                   object
4. Occupation Status                                                                                                     object
5. What type of organizations are you affiliated with?                                                                   object
6. Do you use social media?                                                                                              object
7. What social media platforms do you commonly use?                                                                      object
8. What is the average time you spend on social media every day?                                                         object
9. How often do you find yourself using Social media without a specific purpose?                                          int64
10. How often do you get distracted by Social media when you are busy doing something?                                    int64
11. Do you feel restless if you haven't used Social media in a while?                                                     int64
12. On a scale of 1 to 5, how easily distracted are you?                                                                  int64
13. On a scale of 1 to 5, how much are you bothered by worries?                                                           int64
14. Do you find it difficult to concentrate on things?                                                                    int64
15. On a scale of 1-5, how often do you compare yourself to other successful people through the use of social media?      int64
16. Following the previous question, how do you feel about these comparisons, generally speaking?                         int64
17. How often do you look to seek validation from features of social media?                                               int64
18. How often do you feel depressed or down?                                                                              int64
19. On a scale of 1 to 5, how frequently does your interest in daily activities fluctuate?                                int64
20. On a scale of 1 to 5, how often do you face issues regarding sleep?                                                   int64
dtype: object

3. Mengubah Data

In [ ]:
results = []

cols = df.select_dtypes(include=['float64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom                 Persentase Outliers
1. What is your age?            17.463617

In [ ]:
columns_to_impute = ["1. What is your age?"]

for col in columns_to_impute:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Menggunakan .loc[] agar tidak muncul SettingWithCopyWarning
    df.loc[:, col] = df[col].clip(lower=lower_bound, upper=upper_bound)

In [ ]:
columns_to_impute = ["9. How often do you find yourself using Social media without a specific purpose?"]

for col in columns_to_impute:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Menggunakan .loc[] agar tidak muncul SettingWithCopyWarning
    df.loc[:, col] = df[col].clip(lower=lower_bound, upper=upper_bound)

<ipython-input-54-723ebdb9d8c6>:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '[5.  4.  3.  4.  3.  4.  4.  5.  5.  1.5 5.  4.  3.  5.  5.  3.  2.  5.
 3.  2.  4.  3.  3.  3.  3.  4.  4.  2.  3.  4.  4.  3.  3.  4.  5.  3.
 5.  2.  4.  4.  1.5 3.  1.5 3.  5.  4.  5.  4.  3.  3.  4.  5.  3.  3.
 4.  2.  3.  3.  4.  1.5 4.  2.  4.  3.  2.  3.  5.  2.  4.  4.  2.  5.
 4.  5.  4.  2.  4.  4.  4.  2.  4.  4.  5.  3.  4.  4.  3.  4.  5.  4.
 3.  4.  4.  4.  4.  5.  5.  4.  4.  4.  3.  5.  3.  4.  2.  2.  3.  1.5
 3.  4.  3.  4.  4.  2.  2.  5.  5.  5.  5.  4.  2.  4.  4.  4.  5.  4.
 1.5 3.  5.  3.  2.  4.  3.  3.  2.  5.  3.  3.  4.  4.  3.  4.  4.  3.
 4.  4.  4.  3.  3.  4.  3.  5.  3.  1.5 5.  2.  1.5 3.  3.  5.  4.  3.
 4.  3.  3.  2.  5.  2.  2.  2.  5.  4.  5.  4.  4.  4.  3.  5.  3.  3.
 5.  4.  3.  3.  5.  3.  4.  4.  5.  5.  3.  3.  3.  5.  4.  3.  5.  3.
 2.  3.  3.  4.  5.  4.  5.  3.  4.  5.  3.  2.  3.  

In [ ]:
columns_to_impute = ["11. Do you feel restless if you haven't used Social media in a while?"]

for col in columns_to_impute:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Menggunakan .loc[] agar tidak muncul SettingWithCopyWarning
    df.loc[:, col] = df[col].clip(lower=lower_bound, upper=upper_bound)

<ipython-input-55-f1553b134d3e>:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '[2.  2.  1.  1.  4.  2.  2.  3.  3.  1.  2.  2.  2.  2.  4.5 2.  1.  4.
 2.  3.  2.  4.  3.  2.  4.5 3.  3.  4.  3.  4.5 3.  3.  3.  3.  4.5 4.5
 4.  1.  4.5 2.  1.  2.  1.  4.  1.  4.  2.  1.  3.  1.  3.  4.5 3.  1.
 1.  4.5 1.  2.  2.  1.  3.  1.  3.  1.  1.  4.  4.5 2.  2.  3.  3.  4.5
 2.  1.  3.  1.  3.  4.  3.  4.  4.  2.  3.  4.5 3.  2.  3.  2.  2.  4.
 2.  4.  3.  3.  3.  1.  2.  1.  4.  2.  3.  3.  3.  4.  4.  4.  2.  1.
 1.  3.  4.  4.  4.  3.  4.  4.  4.5 4.5 4.5 3.  1.  1.  4.5 4.  3.  3.
 3.  4.  1.  2.  1.  2.  2.  3.  1.  3.  3.  1.  2.  4.  2.  2.  1.  1.
 4.  3.  3.  3.  3.  2.  1.  4.  2.  2.  1.  1.  2.  1.  2.  4.  1.  1.
 4.5 3.  3.  2.  1.  2.  1.  2.  4.5 4.  2.  4.  4.5 3.  1.  3.  4.  4.5
 4.  2.  1.  4.5 4.  3.  2.  2.  4.  4.5 2.  3.  2.  3.  4.  1.  2.  3.
 2.  3.  2.  2.  3.  3.  4.5 3.  4.  4.5 1.  1.  3.

In [ ]:
columns_to_impute = ["16. Following the previous question, how do you feel about these comparisons, generally speaking?"]

for col in columns_to_impute:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Menggunakan .loc[] agar tidak muncul SettingWithCopyWarning
    df.loc[:, col] = df[col].clip(lower=lower_bound, upper=upper_bound)

<ipython-input-56-0e4eac159f39>:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '[3.  1.  3.  1.  3.  4.  3.  3.  3.  1.  2.  2.  2.  3.  4.5 2.  3.  4.
 3.  4.5 2.  2.  3.  2.  1.  4.  3.  4.  3.  4.  2.  2.  3.  2.  3.  3.
 3.  3.  3.  3.  1.  4.  3.  2.  3.  3.  4.  2.  3.  2.  3.  2.  4.5 3.
 3.  3.  3.  3.  3.  3.  2.  2.  3.  3.  3.  3.  2.  3.  4.  2.  1.  4.5
 2.  3.  2.  1.  3.  1.  3.  2.  3.  3.  3.  4.  2.  2.  2.  4.  3.  2.
 3.  3.  3.  2.  3.  1.  3.  2.  3.  1.  2.  1.  3.  3.  4.  4.  3.  3.
 1.  3.  3.  4.  3.  3.  1.  3.  1.  3.  4.5 3.  3.  2.  3.  1.  4.5 2.
 3.  1.  1.  4.  4.  2.  4.  1.  4.  3.  3.  1.  3.  4.  4.  2.  3.  3.
 2.  2.  1.  2.  3.  4.  4.  4.5 3.  1.  3.  3.  3.  1.  2.  4.5 2.  3.
 3.  3.  4.5 3.  1.  4.5 3.  3.  3.  2.  2.  3.  4.  2.  3.  3.  2.  2.
 2.  3.  3.  4.  3.  2.  3.  3.  3.  4.5 3.  4.5 2.  3.  3.  3.  4.  4.
 2.  3.  2.  4.  3.  2.  3.  2.  2.  4.  1.  4.  4.  

In [ ]:
columns_to_impute = ["12. On a scale of 1 to 5, how easily distracted are you?"]
for col in columns_to_impute:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Menggunakan .loc[] agar tidak muncul SettingWithCopyWarning
    df.loc[:, col] = df[col].clip(lower=lower_bound, upper=upper_bound)

<ipython-input-57-bd7a1545fc74>:10: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '[5.  4.  2.  3.  4.  3.  2.  3.  3.  1.5 4.  5.  4.  1.5 5.  3.  2.  4.
 4.  3.  3.  2.  3.  4.  5.  3.  3.  4.  4.  5.  3.  3.  3.  3.  5.  4.
 3.  1.5 4.  4.  1.5 3.  1.5 4.  3.  4.  2.  3.  3.  3.  4.  5.  5.  3.
 2.  4.  2.  2.  4.  2.  4.  3.  5.  3.  5.  3.  5.  3.  3.  3.  2.  5.
 2.  3.  3.  3.  4.  4.  3.  3.  5.  5.  5.  5.  4.  3.  4.  5.  2.  5.
 4.  5.  4.  4.  2.  3.  4.  2.  4.  5.  4.  4.  3.  3.  3.  3.  4.  1.5
 3.  3.  4.  4.  4.  3.  4.  5.  5.  4.  5.  4.  3.  2.  5.  5.  3.  5.
 3.  3.  2.  3.  3.  3.  3.  2.  2.  3.  3.  3.  3.  5.  2.  5.  2.  1.5
 3.  3.  4.  4.  3.  2.  1.5 3.  3.  3.  2.  5.  5.  2.  2.  3.  2.  1.5
 5.  3.  5.  3.  2.  3.  2.  4.  4.  4.  3.  4.  5.  2.  1.5 4.  5.  2.
 3.  3.  2.  4.  5.  3.  4.  3.  3.  5.  2.  3.  4.  3.  4.  1.5 2.  2.
 2.  2.  5.  2.  3.  5.  5.  2.  5.  5.  1.5 2.  3.

In [ ]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom                                               Persentase Outliers
1. What is your age?                                                0.0
9. How often do you find yourself using Social ...                  0.0
10. How often do you get distracted by Social m...                  0.0
11. Do you feel restless if you haven't used So...                  0.0
12. On a scale of 1 to 5, how easily distracted...                  0.0
13. On a scale of 1 to 5, how much are you both...                  0.0
14. Do you find it difficult to concentrate on ...                  0.0
15. On a scale of 1-5, how often do you compare...                  0.0
16. Following the previous question, how do you...                  0.0
17. How often do you look to seek validation fr...                  0.0
18. How often do you feel depressed or down?                        0.0
19. On a scale of 1 to 5, how frequently does y...                  0.0
20. On a scale of 1 to 5, how often do you face...                  0.0

In [ ]:
df

Timestamp  1. What is your age? 2. Gender  \
0     4/18/2022 19:18:47                  21.0      Male   
1     4/18/2022 19:19:28                  21.0    Female   
2     4/18/2022 19:25:59                  21.0    Female   
3     4/18/2022 19:29:43                  21.0    Female   
4     4/18/2022 19:33:31                  21.0    Female   
..                   ...                   ...       ...   
476   5/21/2022 23:38:28                  24.0      Male   
477    5/22/2022 0:01:05                  26.0    Female   
478   5/22/2022 10:29:21                  29.0    Female   
479   7/14/2022 19:33:47                  21.0      Male   
480  11/12/2022 13:16:50                  33.5      Male   

    3. Relationship Status 4. Occupation Status  \
0        In a relationship   University Student   
1                   Single   University Student   
2                   Single   University Student   
3                   Single   University Student   
4                   Single   University Student   
..                     ...                  ...   
476                 Single      Salaried Worker   
477                Married      Salaried Worker   
478                Married      Salaried Worker   
479                 Single   University Student   
480                Married      Salaried Worker   

    5. What type of organizations are you affiliated with?  \
0                                           University       
1                                           University       
2                                           University       
3                                           University       
4                                           University       
..                                                 ...       
476                                University, Private       
477                                         University       
478                                         University       
479                                         University       
480                                            Private       

    6. Do you use social media?  \
0                           Yes   
1                           Yes   
2                           Yes   
3                           Yes   
4                           Yes   
..                          ...   
476                         Yes   
477                         Yes   
478                         Yes   
479                         Yes   
480                         Yes   

    7. What social media platforms do you commonly use?  \
0    Facebook, Twitter, Instagram, YouTube, Discord...    
1    Facebook, Twitter, Instagram, YouTube, Discord...    
2              Facebook, Instagram, YouTube, Pinterest    
3                                  Facebook, Instagram    
4                         Facebook, Instagram, YouTube    
..                                                 ...    
476                       Facebook, Instagram, YouTube    
477                                  Facebook, YouTube    
478                                  Facebook, YouTube    
479  Facebook, Twitter, Instagram, YouTube, Discord...    
480                                  Facebook, YouTube    

    8. What is the average time you spend on social media every day?  \
0                                Between 2 and 3 hours                 
1                                    More than 5 hours                 
2                                Between 3 and 4 hours                 
3                                    More than 5 hours                 
4                                Between 2 and 3 hours                 
..                                                 ...                 
476                              Between 2 and 3 hours                 
477                              Between 1 and 2 hours                 
478                              Between 2 and 3 hours                 
479                              Between 2 and 3 hours                 
480           

5. Data Reduction

Type of organizations saya rasa tidak memiliki pengaruh langsung terhadap variabel yang ingin saya Aanalisis atau modelkan (seperti kesehatan mental atau penggunaan media sosial)

In [ ]:
df = df.drop('5. What type of organizations are you affiliated with?', axis=1)

In [ ]:
df

Timestamp  1. What is your age? 2. Gender  \
0     4/18/2022 19:18:47                  21.0      Male   
1     4/18/2022 19:19:28                  21.0    Female   
2     4/18/2022 19:25:59                  21.0    Female   
3     4/18/2022 19:29:43                  21.0    Female   
4     4/18/2022 19:33:31                  21.0    Female   
..                   ...                   ...       ...   
476   5/21/2022 23:38:28                  24.0      Male   
477    5/22/2022 0:01:05                  26.0    Female   
478   5/22/2022 10:29:21                  29.0    Female   
479   7/14/2022 19:33:47                  21.0      Male   
480  11/12/2022 13:16:50                  33.5      Male   

    3. Relationship Status 4. Occupation Status 6. Do you use social media?  \
0        In a relationship   University Student                         Yes   
1                   Single   University Student                         Yes   
2                   Single   University Student                         Yes   
3                   Single   University Student                         Yes   
4                   Single   University Student                         Yes   
..                     ...                  ...                         ...   
476                 Single      Salaried Worker                         Yes   
477                Married      Salaried Worker                         Yes   
478                Married      Salaried Worker                         Yes   
479                 Single   University Student                         Yes   
480                Married      Salaried Worker                         Yes   

    7. What social media platforms do you commonly use?  \
0    Facebook, Twitter, Instagram, YouTube, Discord...    
1    Facebook, Twitter, Instagram, YouTube, Discord...    
2              Facebook, Instagram, YouTube, Pinterest    
3                                  Facebook, Instagram    
4                         Facebook, Instagram, YouTube    
..                                                 ...    
476                       Facebook, Instagram, YouTube    
477                                  Facebook, YouTube    
478                                  Facebook, YouTube    
479  Facebook, Twitter, Instagram, YouTube, Discord...    
480                                  Facebook, YouTube    

    8. What is the average time you spend on social media every day?  \
0                                Between 2 and 3 hours                 
1                                    More than 5 hours                 
2                                Between 3 and 4 hours                 
3                                    More than 5 hours                 
4                                Between 2 and 3 hours                 
..                                                 ...                 
476                              Between 2 and 3 hours                 
477                              Between 1 and 2 hours                 
478                              Between 2 and 3 hours                 
479                              Between 2 and 3 hours                 
480                                  Less than an Hour                 

     9. How often do you find yourself using Social media without a specific purpose?  \
0                                                  5.0                                  
1                                                  4.0                                  
2                                                  3.0                                  
3                                                  4.0                                  
4                                                  3.0                                  
..                                                 ...                                  
476                                                3.0                                  
477                                                2.0     